appside/render_body_frames.js

```diff
@@ -1,9 +1,36 @@
 "use strict";
 
+// test method for rendering any of several kinds of frame, specified in query string.
 $(document).ready(function() {
-    statements_frame_main();
+    let test_methods = {
+        'statements': statements_frame_main,
+        'words': words_frame_main,
+    };
+    let page_types = Object.keys(test_methods);
+    let page_to_show = page_types[0];
+
+    // see if a frame type was written in the query string, otherwise use default
+    let query_string = location.search;
+    if (query_string.length > 0) {
+        let query = query_string.substring(1, query_string.length);
+        if (page_types.includes(query)) {
+            page_to_show = query;
+        }
+    }
+
+    let test_fcn = test_methods[page_to_show];
+    test_fcn();
```

Comment:
I assume this is a function to test that it will render the page that is specified in page_to_show? If so, where is test_fcn() written or is this something that will be later implemented?

Reply:
Good question!

`test_fcn` is a variable that holds a function.

The mapping `test_methods` maps strings to functions. We grab text from the query string of the URL (query string is the question mark and everything after it). We use that text to index into our map and figure out which function to call. That function gets assigned to the variable `test_fcn`, which gets called right away.

For example, the URL `file:///home/leah/work/emotional_clarity/appside/test_frames.html?words` has query string `?words`. Since `'words'` is mapped to `words_frame_main`, `test_fcn` gets assigned to `words_frame_main`, and called. 

Glad to discuss this more (including in person) if I haven't sufficiently described it above.

Reply:
Thorough explanation. 👍 Thank you!